feature_engine/encoding/mean_encoding.py

```diff
@@ -127,6 +128,7 @@ def fit(self, X: pd.DataFrame, y: pd.Series):
             The target.
         """
 
+        X, y = _check_X_y_pd_np_mismatch(X, y)
         X = self._check_fit_input_and_variables(X)
```

Comment:
I would remove _is_dataframe from _check_fit_input_and_variables(X) which would now return self

this way we don't carry on the same process twice (check that X is a dataframe)